lib/internal/url.js

```diff
@@ -874,32 +871,32 @@ class URLSearchParams {
     this[context] = null;
   }
 
-  [util.inspect.custom](recurseTimes, ctx) {
+  [util.inspect.custom](depth, opts) {
     if (!this || !this[searchParams] || this[searchParams][searchParams]) {
       throw new TypeError('Value of `this` is not a URLSearchParams');
     }
 
-    if (typeof recurseTimes === 'number' && recurseTimes < 0)
-      return ctx.stylize('[Object]', 'special');
+    if (typeof depth === 'number' && depth < 0)
+      return opts.stylize('[Object]', 'special');
 
-    const separator = ', ';
-    const innerOpts = Object.assign({}, ctx);
-    if (recurseTimes !== null) {
-      innerOpts.depth = recurseTimes - 1;
+    var separator = ', ';
```

Comment:
@benjamingr It's not just this one line though. Many times all `const`/`let` has to be changed before the perm deopt goes away.

As far as checking for permanent deopts, the method I described works well enough:

`node --trace-opt --trace-deopt foo.js | grep -iP 'abort|disabled'`

If anything shows up, then there is a perm deopt somewhere. The name of the function or variable is included in the output FWIW.